subsys/usb/class/hid/core.c

```diff
@@ -675,6 +675,19 @@ static void hid_interface_config(struct usb_desc_header *head,
 		.endpoint = hid_ep_data_##x,				\
 	};
 
+int usb_hid_set_proto_code(const struct device *dev, uint8_t proto_code)
+{
+	struct usb_cfg_data *cfg = (void *)dev->config;
+	struct usb_if_descriptor *if_desc = (void *)cfg->interface_descriptor;
```

Comment:
Do we need the explicit casts here ?

Reply:
ohh, copy & paste from above, but `void *interface_descriptor;` does not need to have const in struct usb_cfg_data().

Reply:
fixed it for this patch.